src/kadmin/dbutil/dump.c

```diff
@@ -688,6 +688,10 @@ process_tl_data(const char *fname, FILE *filep, int lineno,
                      _("cannot read tagged data type and length"));
             return EINVAL;
         }
+        if (i1 > KRB5_INT16_MAX || u1 > KRB5_INT16_MAX) {
+            load_err(fname, lineno, ("overflow"));
```

Comment:
i1 should be compared against INT16_MIN and INT16_MAX.  u1 should be compared against UINT16_MAX.  KRB5_INT16_MAX is defined to a bogus value (old technical debt) and shouldn't be used.  (Same applies to the next few hunks.)